lib/puma/minissl.rb

```diff
@@ -22,6 +33,46 @@ def closed?
         @socket.closed?
       end
 
+      # returns a two element array
+      # first is protocol version (SSL_get_version)
+      # second is 'handshake' state (SSL_state_string)
+      #
+      # used for dropping tcp connections to ssl
+      # see OpenSSL ssl/ssl_stat.c SSL_state_string for info
+      #
+      def ssl_version_state
+        ::Puma::IS_JRUBY ? ['', ''] : @engine.ssl_vers_st.map(&:rstrip)
```

Comment:
This is because you didn't implement a JRuby version, right? Not because the JRuby SSL version should actually return different values?

Reply:
Or our JRuby extension doesn't support TLS 1.3?

Reply:
I wrote the code quite a while ago.  At the time, JRuby didn't support TLSv1.3 yet, and and I opened an issue somewhere asking about it, didn't really get a response.

Reply:
Ah, got it.

Reply:
@MSP-Greg is this the issue? https://github.com/jruby/jruby-openssl/issues/184

Reply:
@dentarg Yes